### PR TITLE
Adding custom Namespacing to the operator

### PR DESCRIPTION
**Pull request by mew2057**
The following Pull Request is engineered to introduce a custom namespace to the operator on creation.

Following changes have been made:
* Added `deploy/namespace.yaml`
* Placed all resources under the `ibm-spectrum-scale-csi-driver`  namespace
* Dropped `Role` for operator.
* Renamed roles to be more descriptive.
* Added a product label.
* `primaryFS` => `primaryFs` 

### TODO

- [x] Clear Molecule
- [x] Test OLM Deployment: https://github.com/operator-framework/community-operators/blob/master/docs/testing-operators.md#testing-operator-deployment-on-kubernetes

### CODE REVIEW

**Comment by mew2057** [comment]
Right now this is breaking `scorecard` and vanilla `molecule`, investigating.

**Comment by mew2057** [comment]
I'm still not confident that hard coding the namespace in the `operator.yaml`, `role.yaml`, `role_binding.yaml` and `service_account.yaml` is correct. I feel like this is why `scorecard` is failing.

**Comment by mew2057** [comment]
Okay, for `scorecard` we at least need to run `kubectl create -f deploy/namespace.yaml`.

**Comment by mew2057** [comment]
To clarify to run the scorecard now we need to run:
``` bash
kubectl create -f deploy/namespace.yaml
operator-sdk scorecard --config .osdk-scorecard.yaml
```

**Comment by mew2057** [comment]
Got `scorecard` running with above recipe, we have errors:

```
{
  "metadata": {
    "creationTimestamp": null
  },
  "log": "time=\"2019-11-13T08:02:19-08:00\" level=info msg=\"Using config file: .osdk-scorecard.yaml\"\ntime=\"2019-11-13T08:04:54-08:00\" level=error msg=\"Plugin `OLM Integration` failed with error (failed to create namespaced resources: failed to get proxyPod: timed out waiting for the condition)\"\n",
  "results": [
    {
      "name": "Basic Tests",
      "description": "Test suite that runs basic, functional operator tests",
      "error": 0,
      "pass": 1,
      "partialPass": 0,
      "fail": 2,
      "totalTests": 3,
      "totalScorePercent": 42,
      "tests": [
        {
          "state": "fail",
          "name": "Writing into CRs has an effect",
          "description": "A CR sends PUT/POST requests to the API server to modify resources in response to spec block changes",
          "earnedPoints": 0,
          "maximumPoints": 1,
          "suggestions": [
            "The operator should write into objects to update state. No PUT or POST requests from the operator were recorded by the scorecard."
          ],
          "errors": []
        },
        {
          "state": "pass",
          "name": "Spec Block Exists",
          "description": "Custom Resource has a Spec Block",
          "earnedPoints": 1,
          "maximumPoints": 1,
          "suggestions": [],
          "errors": []
        },
        {
          "state": "fail",
          "name": "Status Block Exists",
          "description": "Custom Resource has a Status Block",
          "earnedPoints": 0,
          "maximumPoints": 1,
          "suggestions": [
            "Add a 'status' field to your Custom Resource"
          ],
          "errors": []
        }
      ],
      "log": "time=\"2019-11-13T08:02:19-08:00\" level=info msg=\"Running for cr: deploy/crds/ibm_v1alpha1_csiscaleoperator_cr.yaml\"\n\n---\n"
    },
    {
      "name": "OLM Integration",
      "description": "",
      "error": 1,
      "pass": 0,
      "partialPass": 0,
      "fail": 0,
      "totalTests": 0,
      "totalScorePercent": 0,
      "tests": null,
      "log": "failed to create namespaced resources: failed to get proxyPod: timed out waiting for the condition:\nLogs: "
    }
  ]
}```

**Comment by mew2057** [comment]
Ran this by hand and the driver pods failed to come up:
```
failed: [localhost](item = {
	'name': 'csi-plugin.yaml.j2'
}) => {
	"ansible_loop_var": "item",
	"changed": false,
	"error": 422,
	"item": {
		"name": "csi-plugin.yaml.j2"
	},
	"msg": "Failed to create object: b'{\"kind\":\"Status\",\"apiVersion\":\"v1\",\"metadata\":{},\"status\":\"Failure\",\"message\":\"DaemonSet.apps \\\\\"ibm-spectrum-scale-csi\\\\\" is invalid: [spec.template.spec.volumes[0].hostPath.path: Required value, spec.template.spec.containers[0].volumeMounts[0].name: Not found: \\\\\"plugin-dir\\\\\", spec.template.spec.containers[1].volumeMounts[0].name: Not found: \\\\\"plugin-dir\\\\\"]\",\"reason\":\"Invalid\",\"details\":{\"name\":\"ibm-spectrum-scale-csi\",\"group\":\"apps\",\"kind\":\"DaemonSet\",\"causes\":[{\"reason\":\"FieldValueRequired\",\"message\":\"Required value\",\"field\":\"spec.template.spec.volumes[0].hostPath.path\"},{\"reason\":\"FieldValueNotFound\",\"message\":\"Not found: \\\\\"plugin-dir\\\\\"\",\"field\":\"spec.template.spec.containers[0].volumeMounts[0].name\"},{\"reason\":\"FieldValueNotFound\",\"message\":\"Not found: \\\\\"plugin-dir\\\\\"\",\"field\":\"spec.template.spec.containers[1].volumeMounts[0].name\"}]},\"code\":422}\\n'",
	"reason": "Unprocessable Entity",
	"status": 422
}
```

**Comment by mew2057** [comment]
Might be a mismatch on the underlying driver.

**Comment by mew2057** [comment]
The above error was a merge error, now resolved.

**Comment by mew2057** [comment]
@yadaven did you save this work to a git repo? The conflicts from applying the patch I think   are accounting for a large amount of headache here.

**Comment by mew2057** [comment]
I looked at the source of the scorecard to understand why it's failing and it looks like the status check is just a blind check on the status of the CR. I'm beginning to wonder if killing the Namespaced Role is having something to do with this?

**Comment by mew2057** [comment]
```
kubectl get csiscaleoperator.scale.ibm.com/ibm-spectrum-scale-csi -n ibm-spectrum-scale-csi-driver -o yaml
apiVersion: scale.ibm.com/v1alpha1
kind: CSIScaleOperator
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"scale.ibm.com/v1alpha1","kind":"CSIScaleOperator","metadata":{"annotations":{},"name":"ibm-spectrum-scale-csi","namespace":"ibm-spectrum-scale-csi-driver"},"spec":{"clusters":[{"id":"11832033572270148010","primary":{"primaryFs":"fs1","primaryFset":"csiFset2"},"restApi":[{"guiHost":"jdunham-rh72-k8s-scale-master.fyre.ibm.com"}],"secrets":"secret1","secureSslMode":false}],"scaleHostpath":"/ibm/fs1"},"status":{}}
  creationTimestamp: "2019-11-13T18:26:31Z"
  generation: 1
  name: ibm-spectrum-scale-csi
  namespace: ibm-spectrum-scale-csi-driver
  resourceVersion: "550745"
  selfLink: /apis/scale.ibm.com/v1alpha1/namespaces/ibm-spectrum-scale-csi-driver/csiscaleoperators/ibm-spectrum-scale-csi
  uid: 8854bb83-356c-4e14-ab9c-cf6c2b997dfa
spec:
  clusters:
  - id: "11832033572270148010"
    primary:
      primaryFs: fs1
      primaryFset: csiFset2
    restApi:
    - guiHost: jdunham-rh72-k8s-scale-master.fyre.ibm.com
    secrets: secret1
    secureSslMode: false
  scaleHostpath: /ibm/fs1
status:
  conditions:
  - lastTransitionTime: "2019-11-13T18:26:30Z"
    message: Running reconciliation
    reason: Running
    status: "True"
    type: Running
```

**Comment by yadaven** [comment]
Can we try removing namespace from ibm-spectrum-scale-csi-operator-role, Since it is a clusterRole I think it should be cluster wide instead of namespaced.

**Comment by mew2057** [comment]
In the past  this usually indicates that a service account  doesn't have either a  correct entry in `ClusterRole` or `Role`

**Comment by mew2057** [comment]
It looks like `scorecard` was setting the  namespace correctly.

```[root ibm-spectrum-scale-csi-operator]# kubectl get serviceaccount ibm-spectrum-scale-csi-operator-sa -o yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: "2019-11-13T19:21:40Z"
  labels:
    app.kubernetes.io/instance: ssco-ibm-spectrum-scale-csi-operator
    app.kubernetes.io/managed-by: operator
    app.kubernetes.io/name: spectrum-scale-csi-operator
    product: ibm-spectrum-scale-csi
  name: ibm-spectrum-scale-csi-operator-sa
  namespace: default
  resourceVersion: "558633"
  selfLink: /api/v1/namespaces/default/serviceaccounts/ibm-spectrum-scale-csi-operator-sa
  uid: ac89023a-87a0-4f75-8c2a-022145a33a06
secrets:
- name: ibm-spectrum-scale-csi-operator-sa-token-khgdh```

**Comment by mew2057** [comment]
I think the problem is in the  role binding:

```
kubectl get rolebindings ibm-spectrum-scale-csi-operator-binding -o  yaml
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  creationTimestamp: "2019-11-13T19:34:49Z"
  labels:
    app.kubernetes.io/instance: ssco-ibm-spectrum-scale-csi-operator
    app.kubernetes.io/managed-by: operator
    app.kubernetes.io/name: spectrum-scale-csi-operator
    product: ibm-spectrum-scale-csi
  name: ibm-spectrum-scale-csi-operator-binding
  namespace: default
  resourceVersion: "560070"
  selfLink: /apis/rbac.authorization.k8s.io/v1/namespaces/default/rolebindings/ibm-spectrum-scale-csi-operator-binding
  uid: d300d3cd-c0c5-4b6d-b7d5-3ae6faf3b3cd
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: ibm-spectrum-scale-csi-operator-role
subjects:
- kind: ServiceAccount
  name: ibm-spectrum-scale-csi-operator-sa
  namespace: ibm-spectrum-scale-csi-driver
```

**Comment by mew2057** [comment]
```apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  labels:
    app.kubernetes.io/instance: ssco-ibm-spectrum-scale-csi-operator
    app.kubernetes.io/managed-by: operator
    app.kubernetes.io/name: spectrum-scale-csi-operator
    product: ibm-spectrum-scale-csi
  name: ibm-spectrum-scale-csi-operator-binding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: ibm-spectrum-scale-csi-operator-role
subjects:
- kind: ServiceAccount
  name: ibm-spectrum-scale-csi-operator-sa
  namespace: default```

This clears the scorecard.

**Comment by mew2057** [comment]
I was able to fix the `scorecard` with the cluster role binding namespace, but I don't think this will work when testing a real application.

I want to get molecule working again too, and then run real world tests.

**Comment by mew2057** [comment]
This  _totally_ sinks the `molecule` tests: 
https://travis-ci.org/mew2057/ibm-spectrum-scale-csi-operator/jobs/611541813

I added the namespace load, but that's not sufficient.

**Comment by mew2057** [comment]
While waiting on scorecard to run I tried a manual deployment and this happened:
```
{"level":"error","ts":1573676483.983626,"logger":"cmd","msg":"Failed to become leader.","error":"pods \"ibm-spectrum-scale-csi-operator-655987696-trzfs\" is forbidden: User \"system:serviceaccount:ibm-spectrum-scale-csi-driver:ibm-spectrum-scale-csi-operator-sa\" cannot get resource \"pods\" in API group \"\" in the namespace \"ibm-spectrum-scale-csi-driver\"","stacktrace":"github.com/go-logr/zapr.(*zapLogger).Error\n\t/root/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128\nmain.main\n\tibm-spectrum-scale-csi-operator/cmd/manager/main.go:163\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203"}

```

**Comment by mew2057** [comment]
Just realized there's a feature in  `scorecard` to change the namespace the plugins are run in.

**Comment by mew2057** [comment]
Now the driver pods are hanging:

```
[root ibm-spectrum-scale-csi-operator]# kubectl get pods -n ibm-spectrum-scale-csi-driver
NAME                                              READY   STATUS              RESTARTS   AGE
ibm-spectrum-scale-csi-attacher-0                 1/1     Running             1          6m52s
ibm-spectrum-scale-csi-ffqk9                      0/2     ContainerCreating   0          6m49s
ibm-spectrum-scale-csi-operator-655987696-82rvx   2/2     Running             0          7m32s
ibm-spectrum-scale-csi-ph5qj                      0/2     ContainerCreating   0          6m49s
ibm-spectrum-scale-csi-provisioner-0              1/1     Running             0          6m51s
[root ibm-spectrum-scale-csi-operator]# kubectl get pods ibm-spectrum-scale-csi-ffqk9
Error from server (NotFound): pods "ibm-spectrum-scale-csi-ffqk9" not found
[root ibm-spectrum-scale-csi-operator]# kubectl get pods ibm-spectrum-scale-csi-ffqk9 -n ibm-spectrum-scale-csi-driver
NAME                           READY   STATUS              RESTARTS   AGE
ibm-spectrum-scale-csi-ffqk9   0/2     ContainerCreating   0          7m7s
[root ibm-spectrum-scale-csi-operator]# kubectl get pods ibm-spectrum-scale-csi-ffqk9 -n ibm-spectrum-scale-csi-driver -o yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2019-11-13T20:45:28Z"
  generateName: ibm-spectrum-scale-csi-
  labels:
    app: ibm-spectrum-scale-csi
    controller-revision-hash: 5f9c759bcf
    pod-template-generation: "1"
    product: ibm-spectrum-scale-csi
  name: ibm-spectrum-scale-csi-ffqk9
  namespace: ibm-spectrum-scale-csi-driver
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: ibm-spectrum-scale-csi
    uid: 0865e2aa-33ad-4ade-ad9b-8c4b64ceabff
  resourceVersion: "567932"
  selfLink: /api/v1/namespaces/ibm-spectrum-scale-csi-driver/pods/ibm-spectrum-scale-csi-ffqk9
  uid: f580905b-0126-4b26-829b-72b55477b7c2
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - jdunham-rh72-k8s-scale-worker1.fyre.ibm.com
  containers:
  - args:
    - --v=5
    - --csi-address=$(ADDRESS)
    - --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
    env:
    - name: ADDRESS
      value: /var/lib/kubelet/plugins/ibm-spectrum-scale-csi/csi.sock
    - name: CHECKSUM
      value: e8baed0af41b105f30b26a40e713fec7ea26b564
    - name: DRIVER_REG_SOCK_PATH
      value: /var/lib/kubelet/plugins/ibm-spectrum-scale-csi/csi.sock
    - name: KUBE_NODE_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: quay.io/k8scsi/csi-node-driver-registrar:v1.0.1
    imagePullPolicy: IfNotPresent
    name: driver-registrar
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/lib/kubelet/plugins/ibm-spectrum-scale-csi
      name: plugin-dir
    - mountPath: /registration
      name: registration-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ibm-spectrum-scale-csi-node-sa-token-n5vnj
      readOnly: true
  - args:
    - --nodeid=$(NODE_ID)
    - --endpoint=$(CSI_ENDPOINT)
    - --v=5
    - --drivername=ibm-spectrum-scale-csi
    env:
    - name: NODE_ID
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    - name: CHECKSUM
      value: e8baed0af41b105f30b26a40e713fec7ea26b564
    - name: CSI_ENDPOINT
      value: unix:/var/lib/kubelet/plugins/ibm-spectrum-scale-csi/csi.sock
    - name: SCALE_HOSTPATH
      value: /ibm/fs1
    - name: SKIP_MOUNT_UNMOUNT
      value: "yes"
    - name: SECRET_TRIGGER
      value: "-1"
    image: quay.io/mew2057/ibm-spectrum-scale-csi-driver:v0.9.1
    imagePullPolicy: IfNotPresent
    name: ibm-spectrum-scale-csi
    resources: {}
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        add:
        - SYS_ADMIN
      privileged: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/lib/kubelet/plugins/ibm-spectrum-scale-csi
      name: plugin-dir
    - mountPath: /var/lib/kubelet
      mountPropagation: Bidirectional
      name: pods-mount-dir
    - mountPath: /dev
      name: host-dev
    - mountPath: /var/lib/ibm/config
      name: spectrum-scale-config
    - mountPath: /ibm/fs1
      name: gpfs-classic
    - mountPath: /var/lib/ibm/secret1
      name: secret1
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ibm-spectrum-scale-csi-node-sa-token-n5vnj
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostNetwork: true
  nodeName: jdunham-rh72-k8s-scale-worker1.fyre.ibm.com
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: ibm-spectrum-scale-csi-node-sa
  serviceAccountName: ibm-spectrum-scale-csi-node-sa
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/pid-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/network-unavailable
    operator: Exists
  volumes:
  - hostPath:
      path: /var/lib/kubelet/plugins/ibm-spectrum-scale-csi
      type: DirectoryOrCreate
    name: plugin-dir
  - hostPath:
      path: /var/lib/kubelet/plugins_registry/
      type: Directory
    name: registration-dir
  - hostPath:
      path: /var/lib/kubelet
      type: Directory
    name: pods-mount-dir
  - hostPath:
      path: /dev
      type: ""
    name: host-dev
  - configMap:
      defaultMode: 420
      name: spectrum-scale-config-json
    name: spectrum-scale-config
  - hostPath:
      path: /ibm/fs1
      type: Directory
    name: gpfs-classic
  - name: secret1
    secret:
      defaultMode: 420
      secretName: secret1
  - name: ibm-spectrum-scale-csi-node-sa-token-n5vnj
    secret:
      defaultMode: 420
      secretName: ibm-spectrum-scale-csi-node-sa-token-n5vnj
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2019-11-13T20:45:28Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2019-11-13T20:45:28Z"
    message: 'containers with unready status: [driver-registrar ibm-spectrum-scale-csi]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2019-11-13T20:45:28Z"
    message: 'containers with unready status: [driver-registrar ibm-spectrum-scale-csi]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2019-11-13T20:45:28Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - image: quay.io/k8scsi/csi-node-driver-registrar:v1.0.1
    imageID: ""
    lastState: {}
    name: driver-registrar
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        reason: ContainerCreating
  - image: quay.io/mew2057/ibm-spectrum-scale-csi-driver:v0.9.1
    imageID: ""
    lastState: {}
    name: ibm-spectrum-scale-csi
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        reason: ContainerCreating
  hostIP: 172.16.171.117
  phase: Pending
  podIP: 172.16.171.117
  podIPs:
  - ip: 172.16.171.117
  qosClass: BestEffort
  startTime: "2019-11-13T20:45:28Z"
```

**Comment by mew2057** [comment]
I'm beginning  to think that we can't modify the namespace of the operator anymore with these changes, because the molecule test case relies on changing the namespace, and it looks like we can't successfully change the deployment anymore. Additionally, it seems as though trying to change the namespace of service accounts leads to problems.

**Comment by mew2057** [comment]
@yadaven I don't think I'm going to get this tonight, can you take a look at the driver hang and possibly the molecule stuff?

Latest `molecule` run:
https://travis-ci.com/IBM/ibm-spectrum-scale-csi-operator/jobs/256296066

Checking out the code:
```
git checkout -b mew2057-namespace dev
git pull https://github.com/mew2057/ibm-spectrum-scale-csi-operator.git namespace
```

Also I split off the `primaryFs` and driver version stuff, because that seemed more blocking than the namespacing (See #52).

**Comment by mew2057** [comment]
One more thought: 

Since olm is going to be our main deployment schema through operatorhub, and that already namespaces, how necessary is this immediately? Maybe this is a `1.0.1` code drop?

**Comment by mew2057** [comment]
Fixed the hang by adding the secrets  to the  namespace. Although it's weird, because the pod wasn't even up yet/never had been up, so how did it know about the secret?

**Comment by yadaven** [comment]
In README lets mention that secrets needs to be created in same namespace in which driver is going to run.

**Comment by mew2057** [comment]
I tried an OLM deploy and it did not go well, it looks like OLM can't set up the service account and can't run the deployment.

``` YAML
status:
  conditions:
  - lastTransitionTime: "2019-11-14T18:48:07Z"
    lastUpdateTime: "2019-11-14T18:48:07Z"
    message: Created new replica set "ibm-spectrum-scale-csi-operator-dc65dcb76"
    reason: NewReplicaSetCreated
    status: "True"
    type: Progressing
  - lastTransitionTime: "2019-11-14T18:48:07Z"
    lastUpdateTime: "2019-11-14T18:48:07Z"
    message: Deployment does not have minimum availability.
    reason: MinimumReplicasUnavailable
    status: "False"
    type: Available
  - lastTransitionTime: "2019-11-14T18:48:07Z"
    lastUpdateTime: "2019-11-14T18:48:07Z"
    message: 'pods "ibm-spectrum-scale-csi-operator-dc65dcb76-" is forbidden: error
      looking up service account marketplace/ibm-spectrum-scale-csi-operator-sa: serviceaccount
      "ibm-spectrum-scale-csi-operator-sa" not found'
    reason: FailedCreate
    status: "True"
    type: ReplicaFailure
  observedGeneration: 2
  unavailableReplicas: 1
```

**Comment by mew2057** [comment]
Okay a few things:
1. I got OLM mostly working, now I'm seeing this error in the driver pods:
```
[root ibm-spectrum-scale-csi-operator]# kubectl logs  -n marketplace ibm-spectrum-scale-csi-4ghml  driver-registrar
I1114 21:04:01.004921       1 main.go:111] Version: v1.0.1-0-g27703026
I1114 21:04:01.007451       1 main.go:118] Attempting to open a gRPC connection with: "/var/lib/kubelet/plugins/ibm-spectrum-scale-csi/csi.sock"
I1114 21:04:01.007480       1 connection.go:69] Connecting to /var/lib/kubelet/plugins/ibm-spectrum-scale-csi/csi.sock
I1114 21:04:01.008010       1 connection.go:96] Still trying, connection is CONNECTING
I1114 21:04:01.008570       1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I1114 21:04:02.009959       1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I1114 21:04:03.052207       1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I1114 21:04:04.228847       1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
I1114 21:04:05.294888       1 connection.go:96] Still trying, connection is TRANSIENT_FAILURE
```

2. It turns out the CSV doesn't use the `ClusterRoleBinding`, `RoleBinding`, or `ServiceAccount`, so we can freely define in those, _**however**_ the `Role` name _must_ be identical to the `ServiceAccount` name.

3. I have a potential workaround for `molecule` where we redefine the namespace for the `ServiceAccount` bind.

**Comment by mew2057** [comment]
```message: 'Error response from daemon: invalid volume specification: ''/:/'':
          invalid mount config for type "bind": invalid specification: destination
          can''t be ''/'''```

**Comment by mew2057** [comment]
OLM is now working in my local build. 
Fresh candidate fix for the `molecule` build incoming.

**Comment by mew2057** [comment]
The `molecule` modification failed, trying to change the namespace to match the namespace set and see if that works.

**Comment by mew2057** [comment]
Awesome, I just need a review of this and we can merge to dev.

File: README.md [hunk 1]
```diff
@@ -141,6 +142,32 @@ spec:
...
+    - key: "scale"
+      value: "true"
+
+  # Node mapping between K8s node and SpectrumScale node, can have multiple
+  # values.
+  nodeMapping:
+    - k8sNode: "node1"
+      spectrumscaleNode: "scaleNode1"
+
   # ----
```
Comment (yadaven) [inline README.md]: Can we also add a note that secret object for GUI should be created within a namespace where driver will run

File: stable/ibm-spectrum-scale-csi-operator-bundle/operators/ibm-spectrum-scale-csi-operator/hacks/csi-scale-destroy.sh [hunk 2]
```diff
@@ -7,3 +7,4 @@ kubectl delete -f deploy/role.yaml
 kubectl delete -f deploy/service_account.yaml
 kubectl delete -f deploy/role_binding.yaml
 kubectl delete -f deploy/crds/ibm_v1alpha1_csiscaleoperator_crd.yaml
+#kubectl delete -f deploy/namespace.yaml
```
Comment (yadaven) [inline stable/ibm-spectrum-scale-csi-operator-bundle/operators/ibm-spectrum-scale-csi-operator/hacks/csi-scale-destroy.sh]: why we have delete in comments, during tear down we should delete namespace also.

File: stable/ibm-spectrum-scale-csi-operator-bundle/operators/ibm-spectrum-scale-csi-operator/roles/csi-scale/templates/csi-plugin-attacher.yaml.j2 [hunk 3]
```diff
@@ -3,25 +3,28 @@ apiVersion: apps/v1
...
   namespace: "{{ namespace }}"
+  labels:
+    product: "{{ productName }}"
 spec:
   serviceName: "{{ opName }}-attacher"
   replicas: 1
   selector:
     matchLabels:
       app: "{{ opName }}-attacher"
+      product: ibm-spectrum-scale-csi
```
Comment (yadaven) [inline stable/ibm-spectrum-scale-csi-operator-bundle/operators/ibm-spectrum-scale-csi-operator/roles/csi-scale/templates/csi-plugin-attacher.yaml.j2]: we should use "{{ productName }}" here ?

**Pull request merged**